version: "3.8"
services:
  api:
    build:
      context: .
      dockerfile: ./Dockerfile
      args:
        development: "True"
    restart: on-failure
    ports:
      - "${FLASK_RUN_PORT}:8001"
    command: "pipenv run flask run --host=${FLASK_RUN_HOST} --port=8001"
    networks:
      - kafka
      - bookstore
    depends_on:
      - db
    env_file:
      - .env
      - .secrets.env
    volumes:
      - ./src/bookstore.log:/bookstore.log
      - ./src/bookstore_errors.log:/bookstore_errors.log
      - ./src/publication_data.log:/publication_data.log
  db:
    image: postgres:11.8
    container_name: "bookstore_database"
    environment:
      - POSTGRES_USER=${DB_USERNAME}
      - POSTGRES_PASSWORD=${DB_PASSWORD}
      - POSTGRES_DB=${DB_NAME}
    ports:
      - "${DB_PORT}:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
      - ./deploy/postgres/local_db:/docker-entrypoint-initdb.d
    env_file:
      - .env
      - .secrets.env
    networks:
      - bookstore

networks:
  kafka:
    external: true
    name: kafka
  bookstore:
    name: bookstore

volumes:
  postgres_data:
